app/utils.py

```diff
@@ -14,11 +14,13 @@ def __init__(self, configuration, request):
         if self.page_size > configuration.max_page_size:
             self.page_size = configuration.max_page_size
 
-    def items(self, query):
-        return query.paginate(self.page, self.page_size, False).items
+    def paginated_data(self, query):
+        data = query.paginate(self.page, self.page_size, False)
+        setattr(data, "per_page", self.configuration.per_page)
+        return data
 
 
-def standardize_response(payload=None, status_code=200):
+def standardize_response(payload=None, status_code=200, paginated_data=None):
```

Comment:
The only thing I'm on the fence about is having `paginated_data` as its own kwarg.  This function used to take like 5 different arguments and it was very unweildy, and @lord-cant-even came in and fixed that by making it just `payload` and `status_code`. The intent of `payload` was that it could have all of the data in that dictionary and not have a whole bunch of arguments to the method.

I'm willing to concede on this if you have an argument for why an additional kwarg is better, but if it's literally all the same to you, I'd prefer to keep this data in `payload` rather than add any additional arguments to `standardize_response`.

Reply:
And if you do decide to change it, can you also change the default for `payload` to be an empty dictionary and delete these lines?
https://github.com/OperationCode/resources_api/blob/89116af17ffc48b17bbd459ec9eb920f3d3b540a/app/utils.py#L35-L36

I didn't realize until just now that these are unnecessary if we just make the default for `payload` an empty dictionary

Reply:
Thanks for the background for `standardize_response` !
The reason I added the pagination details there is because I was trying to find a way to add the pagination details in one place so they could get added for all of the actions that use pagination.

But maybe it makes more sense to add a `pagination_details` function in `utils.py`. We'd pass in `paginated_data` and it would return
```
{
    "page" :  paginated_data.page
    "number_of_pages" : paginated_data.pages
    "records_per_page": paginated_data.per_page
}
```

then I could call that function in all of the actions that use pagination. and then merge that dictionary with the rest of the payload. That way when people add more actions that use pagination they don't have to include each pagination detail individually, they just know they need to call `pagination_details` and add whatever is returned.

What do you think?

Also, sure, I can change the the default for payload to be an empty dict and delete lines 35-36 👍

Reply:
I'm not going to lie, I'm not entirely sure what that would look like but it sounds good to me. Anything that merges that data with the payload satisfies my distaste with the current situation, so I'm down to check it out. Please make those changes and then I'll review it again and we'll iterate til we get it (but I'm pretty sure this next iteration will be fine if I understand your suggestion correctly). Thanks for your contribution 😄